android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java

```diff
@@ -114,7 +134,8 @@ public BlockersInfo() {
     private View mHardwareButtonMenuAnchor;
     private final Map<Integer, BlockersInfo> mTabsStat =
         Collections.synchronizedMap(new HashMap<Integer, BlockersInfo>());
-
+    private Map<String, DisconnectEntity> mDisconnectEntity =
```

Comment:
I see the map is used, but the map is then iterated element by element in `getBlockerCompanyName`. Nowhere is used for fast search of DisconnectEntity. So there are no advantages of using the map here. 

I would suggest to replace it with some linear container as `ArrayList`. 

WDYT?